src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+    gethostname( buf, sizeof( buf ) );
+
+  struct hostent * host = gethostbyname( buf );
+  if( host == NULL ) {
+    FD_LOG_WARNING( ( "unable to resolve host %s", buf ) );
+    return NULL;
+  }
+  /* Convert result to repair address */
+  res->l    = 0;
+  res->addr = ( (struct in_addr *)host->h_addr )->s_addr;
+  int port  = atoi( str + i + 1 );
+  if( ( port > 0 && port < 1024 ) || port > (int)USHORT_MAX ) {
+    FD_LOG_ERR( ( "invalid port number" ) );
+    return NULL;
+  }
+  res->port = htons( (ushort)port );
+
+  return res;
+}
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 4096UL;
+}
+
+FD_FN_PURE static inline ulong
+loose_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED ) {
+  return 1UL * FD_SHMEM_GIGANTIC_PAGE_SZ;
+}
+
+FD_FN_PURE static inline ulong
+scratch_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED) {
+
+  ulong l = FD_LAYOUT_INIT;
+  l = FD_LAYOUT_APPEND( l, alignof(fd_repair_tile_ctx_t), sizeof(fd_repair_tile_ctx_t) );
+  l = FD_LAYOUT_APPEND( l, fd_repair_align(), fd_repair_footprint() );
+  l = FD_LAYOUT_APPEND( l, fd_repair_align(), fd_repair_footprint() );
+  l = FD_LAYOUT_APPEND( l, fd_scratch_smem_align(), fd_scratch_smem_footprint( SMAX ) );
+  l = FD_LAYOUT_APPEND( l, fd_scratch_fmem_align(), fd_scratch_fmem_footprint( SDEPTH ) );
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof(fd_repair_tile_ctx_t) );
+}
+typedef struct __attribute__((packed)) {
+  fd_eth_hdr_t eth[1];
+  fd_ip4_hdr_t ip4[1];
+  fd_udp_hdr_t udp[1];
+} eth_ip_udp_t;
+
+static inline void
+populate_packet_header_template( eth_ip_udp_t * pkt,
+                                 ulong          payload_sz,
+                                 uint           src_ip,
+                                 uchar const *  src_mac,
+                                 ushort         src_port ) {
+  memset( pkt->eth->dst, 0,       6UL );
+  memcpy( pkt->eth->src, src_mac, 6UL );
+  pkt->eth->net_type  = fd_ushort_bswap( FD_ETH_HDR_TYPE_IP );
+
+  pkt->ip4->verihl       = FD_IP4_VERIHL( 4U, 5U );
+  pkt->ip4->tos          = (uchar)0;
+  pkt->ip4->net_tot_len  = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_ip4_hdr_t)+sizeof(fd_udp_hdr_t)) );
+  pkt->ip4->net_frag_off = fd_ushort_bswap( FD_IP4_HDR_FRAG_OFF_DF );
+  pkt->ip4->ttl          = (uchar)64;
+  pkt->ip4->protocol     = FD_IP4_HDR_PROTOCOL_UDP;
+  pkt->ip4->check        = 0U;
+  memcpy( pkt->ip4->saddr_c, &src_ip, 4UL );
+  memset( pkt->ip4->daddr_c, 0,       4UL ); /* varies by shred */
+
+  pkt->udp->net_sport = fd_ushort_bswap( src_port );
+  pkt->udp->net_dport = (ushort)0; /* varies by shred */
+  pkt->udp->net_len   = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_udp_hdr_t)) );
+  pkt->udp->check     = (ushort)0;
+}
+
+static void FD_FN_UNUSED
+send_packet( fd_repair_tile_ctx_t * ctx,
+             uint    ip,
```

Comment:
```suggestion
             uint      dst_ip_addr,
             ushort dst_port,
```

Reply:
Or are these meant to be src? I can't tell from the broken netmux_sig call below